nireports/reportlets/modality/dwi.py

```diff
@@ -0,0 +1,127 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2023 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Visualizations for diffusion MRI data."""
+import numpy as np
+from matplotlib import pyplot as plt
+
+
+def plot_heatmap(
+    data,
+    b_indices,
+    bvals,
+    mask,
+    scalar,
+    scalar_label="DWI-derived scalar (a.u.)",
+    bins=(150, 11),
+    imax=None,
+    sub_size=100000,
+    cmap="YlGn",
+    sigma=None,
+):
+    """Create heatmap plot."""
+    # Round scalar to a single-digit decimal
+    scalar = np.clip(np.round(scalar + 0.005, 1), 0, 1)
+
+    # Prepare data in shells
+    shells = [np.rint(data[mask][..., idx].reshape(-1)) for idx in b_indices]
+
+    if imax is None:
+        imax = np.percentile(shells[0], 75)
+
+    fig, axs = plt.subplots(
+        len(b_indices) + 1,
+        sharex=True,
+        figsize=(20, 1.6 * (len(b_indices) + 1)),
+    )
+    axs[-1].spines[:].set_visible(False)
+    axs[-1].grid(which="minor", color="w", linestyle='-', linewidth=1)
+    for i, shelldata in enumerate(shells):
+        x = shelldata[shelldata < imax]
+        y = np.array([scalar[mask]] * len(b_indices[i])).reshape(-1)[shelldata < imax]
+
+        if sub_size is not None:
+            choice = np.random.choice(range(x.size), size=sub_size)
+            x = x[choice]
+            y = y[choice]
+
+        alpha = y.copy()
+        alpha = alpha - alpha.min()
+        alpha = i + (2 * alpha / alpha.max() - 1.0) * 0.35
```

Comment:
`alpha` doesn't seem to be used anywhere below. Or am I missing something? 

What does this variable represent?